extensions/neo4j/deployment/src/main/java/io/quarkus/neo4j/deployment/Neo4jDriverProcessor.java

```diff
@@ -22,6 +22,16 @@
 
 class Neo4jDriverProcessor {
 
+    boolean hasJTA() {
```

Comment:
nice that it doesn't require JTA :)

I think your approach should work fine, but remember it's theoretically possible that the `Transactional` class is available during the build but not included in the final application (This is unlikely, but I'll share the idea for sake of explaining).

An alternative would be to explicitly query for the presence of the transactions capability being included in the application; this is made available to extensions via the `Capabilities` build item; e.g. see:

https://github.com/quarkusio/quarkus/blob/834a4cb62682efe46581b8861dc26535432779fe/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java#L156-L161

Reply:
Lovely! I was looking for something like that.